sync-core/src/main/java/com/cloudant/sync/query/QueryExecutor.java

```diff
@@ -250,9 +248,7 @@ private boolean validateFields(List<String> fields) {
                 if (accumulator == null) {
                     accumulator = new HashSet<String>(childIds);
                 } else {
-                    if (childIds != null) {
-                        accumulator = Sets.union(accumulator, childIds);
-                    }
+                    accumulator = Sets.union(accumulator, childIds);
```

Comment:
This check really should have been removed when we moved the childIds == null check above the accumulator creation. https://github.com/cloudant/sync-android/commit/a4626ee5d3f29d9899d43e4a01467be78d92bd19#diff-80ec4d1d12847da53428d6c14a88dbb7L227

Basically this check for childIds != null is redundant and needed to be removed.